extension/parquet/reader/variant/variant_binary_decoder.cpp

```diff
@@ -0,0 +1,360 @@
+                                                          const VariantValueMetadata &value_metadata,
+                                                          const_data_ptr_t data) {
+	switch (value_metadata.primitive_type) {
+	case VariantPrimitiveType::NULL_TYPE: {
+		return yyjson_mut_null(doc);
+	}
+	case VariantPrimitiveType::BOOLEAN_TRUE: {
+		return yyjson_mut_true(doc);
+	}
+	case VariantPrimitiveType::BOOLEAN_FALSE: {
+		return yyjson_mut_false(doc);
+	}
+	case VariantPrimitiveType::INT8: {
+		auto value = Load<int8_t>(data);
+		return yyjson_mut_int(doc, value);
+	}
+	case VariantPrimitiveType::INT16: {
+		auto value = Load<int16_t>(data);
+		return yyjson_mut_int(doc, value);
+	}
+	case VariantPrimitiveType::INT32: {
+		auto value = Load<int32_t>(data);
+		return yyjson_mut_int(doc, value);
+	}
+	case VariantPrimitiveType::INT64: {
+		auto value = Load<int64_t>(data);
+		return yyjson_mut_int(doc, value);
+	}
+	case VariantPrimitiveType::DOUBLE: {
+		double value;
+		memcpy(&value, data, sizeof(double));
+		return yyjson_mut_real(doc, value);
+	}
+	case VariantPrimitiveType::DECIMAL4: {
+		uint8_t scale;
+		uint8_t width;
+
+		auto value = DecodeDecimal<int32_t>(data, scale, width);
+		auto value_str = Decimal::ToString(value, width, scale);
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::DECIMAL8: {
+		uint8_t scale;
+		uint8_t width;
+
+		auto value = DecodeDecimal<int64_t>(data, scale, width);
+		auto value_str = Decimal::ToString(value, width, scale);
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::DECIMAL16: {
+		uint8_t scale;
+		uint8_t width;
+
+		auto value = DecodeDecimal<hugeint_t>(data, scale, width);
+		auto value_str = Decimal::ToString(value, width, scale);
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::DATE: {
+		date_t value;
+		value.days = Load<int32_t>(data);
+		auto value_str = Date::ToString(value);
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::TIMESTAMP_MICROS: {
+		timestamp_tz_t micros_tz_ts;
+		micros_tz_ts.value = Load<int64_t>(data);
+
+		auto value = Value::TIMESTAMPTZ(micros_tz_ts);
+		auto value_str = value.ToString();
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::TIMESTAMP_NTZ_MICROS: {
+		timestamp_t micros_ts;
+		micros_ts.value = Load<int64_t>(data);
+
+		auto value = Value::TIMESTAMP(micros_ts);
+		auto value_str = value.ToString();
+		return yyjson_mut_strcpy(doc, value_str.c_str());
+	}
+	case VariantPrimitiveType::FLOAT: {
+		float value;
+		memcpy(&value, data, sizeof(float));
+		return yyjson_mut_real(doc, value);
+	}
+	case VariantPrimitiveType::BINARY:
+	case VariantPrimitiveType::STRING: {
+		auto size = Load<uint32_t>(data);
+		auto string_data = reinterpret_cast<const char *>(data + sizeof(uint32_t));
+		return yyjson_mut_strncpy(doc, string_data, size);
```

Comment:
`STRING` is only allowed to have UTF8 values, so we should probably verify that here